vm/rust/src/lib.rs

```diff
@@ -223,10 +221,13 @@ pub extern "C" fn cairoVMExecute(
             return;
         }
 
+        let mut txn_state = CachedState::create_transactional(&mut state);
```

Comment:
Could you explain why do we need this?

Reply:
to be able to generate per-txn state diffs. 

txn_state buffers the changes internally, until `commit()` is called. So we gather all the changes for a single txn, save them separately and then commit them to `state`.